WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -71,16 +71,16 @@ def index
     @years = ["all years"] + Competition.where(showAtAll: true).pluck(:year).uniq.select { |y| y <= Date.today.year }.sort!.reverse!
     @competitions = Competition.where(showAtAll: true).order(:year, :month, :day)
 
-    if @display != "admin"
-      if @present_selected
-        @competitions = @competitions.where("CAST(CONCAT(year,'-',endMonth,'-',endDay) as Datetime) >= ?", Date.today)
-      else
-        @competitions = @competitions.where("CAST(CONCAT(year,'-',endMonth,'-',endDay) as Datetime) < ?", Date.today).reverse_order
-        unless params[:year] == "all years"
-          @competitions = @competitions.where(year: params[:year])
-        end
-      end
+    if @present_selected
+      @competitions = @competitions.where("CAST(CONCAT(year,'-',endMonth,'-',endDay) as Datetime) >= ?", Date.today)
```

Comment:
Do these just rely on whatever ordering the DB happens to return? MySQL usually (maybe always) return stuff ordered by `id`, but Postgres does **not**. So I'd slap some `.order(:not_sure_what_column)` on those.